### PR TITLE
homework week1

### PR DESCRIPTION
**Pull request by jimfang**
just concept. didn't compile as not set up the environment properly.
Raise Issues for it.

### CODE REVIEW

File: cis194/week1/jimfang/Homework01.hs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+-- toDigits
+toDigits :: Integer -> [Integer]
+toDigits num
+  | num <= 0 = []
+  | otherwise = toDigits ( num `div` 10 ) ++ [ num `mod` 10 ]
+
+-- toDigitRev
+toDigitsRev :: Integer -> [Integer]
+toDigitRev = reverse . toDigits
```
Comment (shineli1984) [inline cis194/week1/jimfang/Homework01.hs]: The function name should be `toDigitsRev`.

File: cis194/week1/jimfang/Homework01.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+-- doubleEveryOther
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther []         = []
+doubleEveryOther [x]        = [x]
+doubleEveryOther [x:y:rest] = [x: y*2] ++ doubleEveryOther rest
```
Comment (shineli1984) [inline cis194/week1/jimfang/Homework01.hs]: The convention is Haskell is to put space before and after `:`
```suggestion
doubleEveryOther [x : y : rest] = [ x : y * 2 ] ++ doubleEveryOther rest
```

File: cis194/week1/jimfang/Homework01.hs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+-- sumDigits
+sumDigits :: [Integer] -> Integer
+sumDigits = sum . concatMap . toDigits
+
+-- validate
+tenTimes :: Integer -> Bool
+tenTimes num = num `mod` 10 == 0
+
+validate :: Integer -> Bool
+validate = tenTimes . sumDigits . doubleEveryOther . toDigits
```
Comment (shineli1984) [inline cis194/week1/jimfang/Homework01.hs]: I think the requirement is doubling every other digit from the right. Is this logic correct?

File: cis194/week1/jimfang/Homework01.hs (hunk 1 above)
Comment (jimfang) [reply cis194/week1/jimfang/Homework01.hs]: thanks

File: cis194/week1/jimfang/Homework01.hs [hunk 4]
```diff
@@ -14,13 +14,16 @@ toDigits num
...
+doubleEveryOtherFromLeft :: [Integer] -> [Integer]
+doubleEveryOtherFromLeft [ ]         = [ ]
+doubleEveryOtherFromLeft [ x ]       = [ x ]
+doubleEveryOtherFromLeft [ x : y : rest ] = [ x : y * 2 ] ++ doubleEveryOther rest
+
 doubleEveryOther :: [Integer] -> [Integer]
-doubleEveryOther []         = []
-doubleEveryOther [x]        = [x]
-doubleEveryOther [x:y:rest] = [x: y*2] ++ doubleEveryOther rest
+doubleEveryOther = reverse . doubleEveryOtherFromLeft . reverse
```
Comment (shineli1984) [inline cis194/week1/jimfang/Homework01.hs]: Nit: We can use toDigitsRev in the sumDigits without changing this function.

**Pull request merged**